tony-core/src/main/java/com/linkedin/tony/TonyClient.java

```diff
@@ -854,6 +889,89 @@ public int start() {
     return -1;
   }
 
+  /**
+   * Look up and return the resource manager's principal. This method
+   * automatically does the <code>_HOST</code> replacement in the principal and
+   * correctly handles HA resource manager configurations.
+   *
+   * @param conf the {@link Configuration} file from which to read the
+   * principal
+   * @return the resource manager's principal string or null if the
+   * {@link YarnConfiguration#RM_PRINCIPAL} property is not set in the
+   * {@code conf} parameter
+   * @throws IOException thrown if there's an error replacing the host name
+   */
+  public static String getRmPrincipal(Configuration conf) throws IOException {
```

Comment:
is this piece of code copied from Hadoop 2.9?

Reply:
Yes, changes are from https://github.com/linkedin/TonY/pull/172.